include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,126 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        /*!
+         * Allows us to avoid code duplication on the attribute creation
+         * function.
+         */
+        template < typename T >
+        static std::string type_to_string()
+        {
+            if( typeid( T ).name() == typeid( index_t ).name() )
+            {
+                return "index_t";
+            }
+            else if( typeid( T ).name() == typeid( int ).name() )
+            {
+                return "int";
+            }
+            else if( typeid( T ).name() == typeid( float ).name() )
+            {
+                return "float";
+            }
+            else if( typeid( T ).name() == typeid( double ).name() )
+            {
+                return "double";
+            }
+            else
+            {
+                throw RINGMeshException(
+                    "CartesianGrid", "Give a valid attribute type." );
```

Comment:
It is not really a question of validity. You can throw a message explaining that the given type is not yet implemented for attribute and it can easily be added in this function/map.

Reply:
Yes it is (more or less), I already implemented all the types that can be used to create a geogram attribute with a single value.